lib/netext/httpext/transport.go

```diff
@@ -116,8 +116,17 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 		}
 	} else {
 		if enabledTags.Has(stats.TagURL) {
-			u := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}
-			tags["url"] = u.Clean()
+			cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
+			tags["url"] = cleanURL
+		}
+
+		if _, ok := tags["name"]; !ok && enabledTags.Has(stats.TagName) {
+			cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
+			tags["name"] = cleanURL
+		}
```

Comment:
To be honest given the high amount of code that will execute those, I would prefer if we do actually do this in (pseudo code):
```
var urlEnabled = enabledTags.Has(url)...;
var nameToBeSet = <the whole if>;
if urlEnabled || nameToBeSet {
  cleanUrl = ...
  if urlEnabled { 
    tags["url"] cleanURL
  } 
  if name....
```
both are likely to be true always and I would argue it isn't the worst code out there

Reply:
Sure, makes sense, see 8ee1d1ca.